### ISSUE TITLE
Missing default connection equation in composite bus connector

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#10502]
Please consider the following MWE:
```modelica
package Expandable_MWEs
  package Interfaces
    expandable connector Bus
    end Bus;
  end Interfaces;

  package Components
    model Gen_pinA "connect an electric pin to bus, act as generator"
      Interfaces.Bus bus;
      model foo
         parameter Real A = 1;
         parameter Real w = 1;
         Modelica.Electrical.Analog.Interfaces.PositivePin a;
      equation
         a.v = A*sin(w*time);
      end foo;
      foo f;
    equation
      connect(f.a, bus.a);
    end Gen_pinA;

    model Usr_pinA "connect an electric pin to bus, act as utiliser"
      Interfaces.Bus bus;
      model foo
      parameter Real R = 1;
         Modelica.Electrical.Analog.Interfaces.PositivePin a;
      equation
         a.v = R*a.i;
      end foo;  
      foo f;
    equation
      connect(bus.a, f.a);
    end Usr_pinA;
  end Components;

  package Examples
    model Example_02_Gen_pinA
      Components.Gen_pinA G;
    end Example_02_Gen_pinA;
  end Examples;
annotation(
    uses(Modelica(version = "4.0.0")));
end Expandable_MWEs;
```
When flattening the Example_02_Gen_pinA model, I get the following flat Modelica code:
```modelica
class Expandable_MWEs.Examples.Example_02_Gen_pinA
  Modelica.Electrical.Analog.Interfaces.PositivePin G.bus.a "virtual variable in expandable connector";
  parameter Real G.f.A = 1.0;
  parameter Real G.f.w = 1.0;
  Real G.f.a.v(quantity = "ElectricPotential", unit = "V") "Potential at the pin";
  Real G.f.a.i(quantity = "ElectricCurrent", unit = "A") "Current flowing into the pin";
equation
  G.f.a.v = G.bus.a.v;
  G.f.a.i - G.bus.a.i = 0.0;
  G.f.a.v = G.f.A * sin(G.f.w * time);
end Expandable_MWEs.Examples.Example_02_Gen_pinA;
```
As we understand from [Section 9.2 of the MLS](https://specification.modelica.org/maint/3.6/connectors-and-connections.html#generation-of-connection-equations)
> Composite connector types are broken down into primitive components. 

Therefore, since `G.bus.a` has no outside connections, we expect the equation `G.bus.a.i = 0` to be generated, which does not happen, making the model under-determined.

**Comment by perost** [comment]
I thought I fixed the issue with #10504, which changes it so that we collect flow variables after handling expandable connectors instead of before (since expandable connectors can add flow variables to the flat model).

However, this isn't actually enough to fix the issue, because we don't handle adding connectors to exandable connectors properly yet. You can see this by the fact that you get `G.bus.a` in the flat model instead of `G.bus.a.v` and `G.bus.a.i` like you should. So we still miss that `G.bus.a.i` is a flow variable since it's never added to the flat model.

**Comment by casella** [comment]
> You can see this by the fact that you get `G.bus.a` in the flat model instead of `G.bus.a.v` and `G.bus.a.i` like you should. So we still miss that `G.bus.a.i` is a flow variable since it's never added to the flat model.

Aha, I missed that. I hope that's not too hard to fix.

**Comment by perost** [comment]
Fixed in #10540.
